src/utils/sources.ts

```diff
@@ -0,0 +1,159 @@
+import {NamePath} from 'antd/lib/form/interface';
+
+import {Option} from '@models/form';
+import {SourceWithRepository} from '@models/sources';
+
+import {notificationCall} from '@molecules';
+
+const customGitSourceString = '$custom-git-';
+
+export const remapTestSources = (testSources: SourceWithRepository[]) => {
+  if (!testSources || !testSources.length) {
+    return [];
+  }
+
```

Comment:
Based on the function definition, `testSources` is always an array, so this part could be omitted.

```suggestion
```